### DISCUSSION TITLE
LLM with /crawl

### DISCUSSION THREAD

**Discussion by justKishore** [firecrawl/firecrawl#697] [Q&A]
Does fire crawl support LLM extraction for multiple routes in a website, i.e. with /crawl functionality?

**Comment by Adit-Jain-srm** [comment]
Yes, Firecrawl supports LLM extraction across multiple routes via `/crawl`. You combine crawling with the `extract` option in `scrapeOptions`:

```python
from firecrawl import FirecrawlApp

app = FirecrawlApp(api_key='your-key')

result = app.crawl_url(
    'https://example.com',
    params={
        'limit': 50,
        'scrapeOptions': {
            'formats': ['extract'],
            'extract': {
                'schema': {
                    'type': 'object',
                    'properties': {
                        'title': {'type': 'string'},
                        'price': {'type': 'number'},
                        'description': {'type': 'string'}
                    }
                }
            }
        }
    }
)

for page in result['data']:
    print(page.get('extract'))  # Structured data per page
```

The crawl will visit multiple pages based on your `limit`, `includePaths`/`excludePaths`, and `maxDepth` settings. Each page gets processed through the LLM extraction pipeline with your schema.

Alternatively, for more targeted multi-page extraction, you can use the dedicated `/extract` endpoint which is optimized for pulling structured data across a site.
